lld/MachO/OutputSegment.cpp

```diff
@@ -89,9 +90,17 @@ static int sectionOrder(OutputSection *osec) {
   StringRef segname = osec->parent->name;
   // Sections are uniquely identified by their segment + section name.
   if (segname == segment_names::text) {
+    if (osec->name == section_names::header)
+      return -7;
+    if (osec->name == section_names::text)
+      return -6;
+    // Ensure all code sections are contiguous with `__text` for thunk
+    // calculations.
+    if (sections::isCodeSection(osec->name, segment_names::text, osec->flags) &&
+        osec->name != section_names::stubHelper) {
+      return -5;
```

Comment:
Super nit:

`isCodeSection` is supposed to return true for `__text`, too, right? Should these two be merged? Or does `__text` need to come first in all cases (in that case, we should clarify that with a comment).

Reply:
This is partly just to be closer to existing behavior, and partly for jankier reasons: we expect `__text` to be the biggest section, so we need it to go first so it can trip `needsThunks`.

I guess the options are:
- Keep it that way and put the above as a comment
- Not encode the assumption and sort code sections by size

WDYT?

Reply:
Comment + keeping the jank sounds okay to me, I don't think we'll hit the case when `__text` is not the largest.

Reply:
Done